SmartTutor/ContentModel/ChallengeService.cs

```diff
@@ -0,0 +1,105 @@
+﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
+using RepositoryCompiler.Controllers;
+using SmartTutor.ContentModel.LearningObjects;
+using SmartTutor.ContentModel.LearningObjects.Repository;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SmartTutor.ContentModel
+{
+    public class ChallengeService : IChallengeService
```

Comment:
We should always keep in mind that the primary job of a service is to coordinate other (smarter) objects to collectively fulfill a use case.